### ISSUE TITLE
upgrade gorethink to v3

### ISSUE THREAD

**Issue by vikstrous** [docker/notary#1073]
gorethink v3 has some fixes that we would like to be able to use, but since we depend on notary, we can't use gorethink v3 ourselves. If both v2 and v3 are used at the same time, it causes a panic in protobufs:
```
panic: proto: duplicate enum registered: VersionDummy_Version
```
Please upgrade to gorethink v3.

**Comment by cyli** [comment]
@vikstrous Does this need to be backported to a particular version series, or can it go into master?

**Comment by cyli** [comment]
@vikstrous Nm, I checked slack and saw that it requires a backport.  :)

**Comment by HuKeping** [comment]
Closes this via #1074 & #1075 , thanks @cyli for picking this up and thanks @vikstrous!
